server/connectors/cloudbees_connector/oc_client.py

```
@@ -0,0 +1,246 @@
+
+    def get_controller_client(self, controller_url: str) -> JenkinsClient:
+        """Return a JenkinsClient configured for a specific managed controller."""
+        self._validate_controller_url(controller_url)
+        return JenkinsClient(
+            base_url=controller_url.rstrip("/"),
+            username=self.username,
+            api_token=self.api_token,
+        )
+
+    def query_recent_builds_across_controllers(
+        self, service: Optional[str] = None, time_window_hours: int = 24
```

Ruff (0.15.15)

[warning] Unused method argument: `time_window_hours`

(ARG002)